tools/codegen/model.py

```diff
@@ -1335,6 +1335,97 @@ def __str__(self) -> str:
         else:
             return f"{self.name}"
 
+@dataclass(frozen=True)
+class ExternalBackendMetadata:
```

Comment:
An alternative data model, which would require less structs to be defined here, is to index the external backend metadata separately from ExternalBackendFunction, rather than insisting it be directly contained here. Then, anywhere you previously would have taken an ExternalBackendFunction, you instead take a NativeFunction and then look up the ExternalBackendMetadata from the index.

Reply:
Yeah, that reduces the amount every data class stuff that goes in here, at the cost of requiring all of the code that acts on ExternalBackendFunctions to do a little more work.

I have a slight preference for the beefed-up data model. The properties I defined on `ExternalBackendFunction` save a fair amount of redundant work from all of the callers, and will abstract any future logic we add to the metadata; e.g. the fact that not having a metadata -> the backend hasn't implemented that op, or that `primary` will figure out if the backend favors `out` vs `functional` operators. But let me know if you feel strongly one way or the other.